### ISSUE TITLE
Error: [Alexa] ERROR: HAP Discovery failed, please review config 

### ISSUE THREAD

**Issue by djcrawleravp** [NorthernMan54/homebridge-alexa#412]
How can i turn my devices to insecure mode? I tried to follow the instruction but didn't find the file.

All I have is Alexa and Pihole as devices, everything works fine but the logs keeps reading this message several times a minute

**Comment by NorthernMan54** [comment]
Do you have any homebridge devices ?

**Comment by djcrawleravp** [comment]
Pihole plugin to enable and disable if it counts

**Comment by NorthernMan54** [comment]
And how are you running homebridge, what instructions did you follow to set it up ?

**Comment by djcrawleravp** [comment]
I think I’ve found the issue, Alexa was discovering itself as a homebridge device on the Alexa App

I disabled it and the message stopped.

Strangely Alexa appears as a Sensor in homekit

**Comment by NorthernMan54** [comment]
That is by design, the sensor is the status of cloud connection for the service

**Comment by miixxii** [comment]
Hello,

I have same error message however my case seems to be different. I'm describing me issue here as I do not want to open to "new issue" as error message is same.

Details about my setup:
- 2 homebridges running via hb-service
-  Homebridge Alexa plugin installed on both
- In both plugins is set to filter Homebridge Alexa to the current one
          One is set to a localhost IP (127.0.0.1: 51715)
          One is set to a private IP (192.168.0.15: 51828)

I took this IPs from debug without a filter

2021-01-22T14:36:07.917Z hapNodeJSClient HAP Device discovered Homebridge 9D18
2021-01-22T14:36:07.920Z hapNodeJSClient Filtered HAP instance address: homebridge -> http://192.168.0.15:51828
2021-01-22T14:36:09.070Z hapNodeJSClient HAP Device discovered Maya 88FC
2021-01-22T14:36:09.071Z hapNodeJSClient Filtered HAP instance address: homebridge -> http://127.0.0.1:51715

If I do not set 2nd filter to localhost (127.0.0.1) by use private IP 192.168.0.15 (same like for other instance), I do have immediately error: ERROR: HAP Discovery failed, please review config

If I use 127.0.0.1 I can manage Homebridge Alexa to be working for some minutes, but after it is again bringing error: ERROR: HAP Discovery failed, please review config, and private IP is shown in log for 2nd instance. 

Here details:

2021-01-22T14:40:35.248Z alexaLocal connect command/********/#
2021-01-22T14:40:36.039Z hapNodeJSClient HAP Device discovered Maya 88FC
2021-01-22T14:40:36.261Z hapNodeJSClient Homebridge instance discovered homebridge with 3 accessories
2021-01-22T14:40:43.443Z hapNodeJSClient HAP Device discovered Homebridge 9D18
2021-01-22T14:40:43.445Z hapNodeJSClient Filtered HAP instance address: homebridge -> http://127.0.0.1:51828
2021-01-22T14:40:53

after some time

2021-01-22T15:10:33.848Z hapNodeJSClient Starting Homebridge instance discovery
2021-01-22T15:10:33.904Z hapNodeJSClient HAP Device discovered Maya 88FC
2021-01-22T15:10:33.905Z hapNodeJSClient Filtered HAP instance address: homebridge -> http://192.168.0.15:51715
2021-01-22T15:10:33.955Z hapNodeJSClient HAP Device discovered Homebridge 9D18
2021-01-22T15:10:33.956Z hapNodeJSClient Filtered HAP instance address: homebridge -> http://192.168.0.15:51828
2021-01-22T15:10:53.849Z hapNodeJSClient Ready
2021-01-22T15:10:53.849Z alexaActions alexaDiscovery
[1/22/2021, 4:10:53 PM] [Alexa-Maya] ERROR: HAP Discovery failed, please review config
2021-01-22T15:10:53.852Z alexaActions registerEvents []

**Comment by NorthernMan54** [comment]
**_Homebridge Alexa plugin installed on both_**

Only install it once, it will discover the second homebridge instance automatically.  Having two plugin instances to the same Amazon account is not supported.

**Comment by miixxii** [comment]
> **_Homebridge Alexa plugin installed on both_**
> 
> Only install it once, it will discover the second homebridge instance automatically. Having two plugin instances to the same Amazon account is not supported.

Hi,

It's installed only once, however plugin is existing on 2 homebridge instances. It's using 2 different amazon accounts, as it's suppose to manage 2 different homebridges. I do not want to mix them together, that's why a filter is used.

**Comment by NorthernMan54** [comment]
Look back at the debug messages carefully, you filtered out all the instances, hence no devices found, hence the error

**Comment by miixxii** [comment]
Yes that’s right, but log is from one instance, the one running on port 51715 so it should not take into account filter from another instance right?  But looks like it’s mixing settings between instances

**Comment by NorthernMan54** [comment]
That error is saying that no devices were found

And the filter needs to match what discovery thinks the instance ip and port is, as that is what it is comparing

**Comment by miixxii** [comment]
If you wish I can send you complete log. If you read my initial post you’ll see that once homebridge is started for instance 1 with filter set to 127.0.0.1:51715 is working for some time. After for I don’t know what reason is start talking about ip 192.168.0.15:51715 (the ip is corrent ip of that server) 

to me is just strange that once homebridge alexa is discovering localhost (ip 127.0.01) and after sometime private ip (192.168.0.15)

**Comment by NorthernMan54** [comment]
Discovery is using mDNS and it is using the homebridge instance address advertised inside.   Why it is doing this I don't know, but it is likely something related to the docker environment you are running in.  Unfortunately I'm not a docker user so really can't comment on what the possible solution to this is.  Your into a non-standard config so not sure how much help I can be here.

The filter option works, and it works well as I use it with my test environment when Amazon is is doing skill validation to stop them from turning all my lights on in the middle of the night.  During the early days of the skill, they had turned on all my devices at 3am, including stereos etc and woke up the whole house.  Very negative WAF score from that.  Since then I implemented the filter to restrict them to a single instance and accessory.
